### DISCUSSION TITLE
Requesting example for live data monitoring

### DISCUSSION THREAD

**Discussion by acrlakshman** [sidbannet/converge_cfd_user#8] [General]
Hi @sidbannet, during your talk at CONVERGE UC, you mentioned that you run CFD jobs on AWS or some other cloud provideres and monitor the results. Do you have any screencast to share?

* I do not have experience with using on the cloud, so if you can provide a screencast or an example, we can discuss together on replicating similar features (if possible) as part of this project.
* By adding more features to `post`, we could build a package via ipywidgets that can group multiple mostly used post-processing steps into one widget and provide it to user via jupyter notebook. In its current form, this project is made of scripts, by adding more features along with classes that compose of widgets, we can make this more mature.

**Comment by sidbannet** [comment]
@acrlakshman I am not sure if I can share screenshot of results data but here is a screenshot of my jobs running at AWS with cloud formation stack called [SOCA](https://github.com/awslabs/scale-out-computing-on-aws)

Screenshot showing m CFD jobs running at AWS is here:

![Capture](https://user-images.githubusercontent.com/60864292/135546653-af7b803a-585d-4bfd-bcc5-73783959d2c0.JPG)

**Reply by acrlakshman** [reply]
@sidbannet oh yes, you might have run mostly company related cases, I thought you might have some toy test cases ran. That' ok, I got the idea. I shall get back to you with some code update and see what you think about future direction.

**Reply by sidbannet** [reply]
Usually I write data in AWS EFS and then use SageMaker (AWS Notebook instance) to read the data from EFS in real time and go through the analysis of the data while the simulation job is still running.

**Reply by acrlakshman** [reply]
That means, if AWS provides an API to talk to their storage filesystem (EFS), then we could do that real time analysis on an extension of this project. We can potentially extend this project as web app where an extension to current scripts can act as backend and a light weight front-end that loads jupyter notebook from this project can talk to user data whether that data be on AWS, GCP, etc. What do you think?

**Comment by sidbannet** [comment]
I like your idea of building out post processing toolset as well. I have it obviously for my work use, and left it out from this repository for quicker adoption of loading the data into `pandas` DataFrame. I agree to build out `post` module with new features. Is that something you are interested in @acrlakshman? Is anyone from CSI want to build it out? Thoughts? Comments? Questions?

**Reply by acrlakshman** [reply]
I do not know about other colleagues, out of interest in oss and trying to build something for CONVERGE users, I did not find enough motivation to push an application until I saw your presentation. This is completely out of self interest, with hope that other users could find it helpful. Also, I don't know how exactly to build a tool that is beneficial to users, since I am not on the support side, I didn't see the pain points first-hand. But after your presentation, I saw this as an opportunity and thought I can contribute to this project and make something out of this that can standout, while seeking your feedback, since you have experience of both on the development side + user side.

**Reply by sidbannet** [reply]
Thanks @acrlakshman I definitely welcome and thank your contribution to this repository. I hope if enough people get interested (which from the messages I have received so far seems yes), this could be a standing topic in CONVERGE User Conference year on year. I would welcome anyone's participation. I think the real pain point is to keep consistent API (file formats, directory structures) across various CONVERGE releases. If the API changes all users have to go through updating their side of scripts to correctly load and analyze CFD data.

**Reply by sidbannet** [reply]
I hope to learn from you as well in this co-development process.

**Reply by sidbannet** [reply]
I am trying to find a way to make you a maintainer for this repo as well. @acrlakshman

**Reply by acrlakshman** [reply]
I think from CONVERGE' side, there is an effort to keep consistent API, but I do not know complete details. However, we can release multiple versions of this project to be in sync with various changes from CONVERGE side too. It sounds more work, but it is a possibility.

**Reply by acrlakshman** [reply]
You can just add me as a collaborator, and depending on my contributions, you may bump my role later.

**Reply by sidbannet** [reply]
I can't see you while trying to invite you as `collaborator`. Is there a personal setting you may need to change? @acrlakshman 
![Capture](https://user-images.githubusercontent.com/60864292/135548199-23268863-f307-4e81-be6b-792630720d0e.JPG)

**Reply by acrlakshman** [reply]
Nothing that I have done to stop this, can you search without `@`, just `acrlakshman` and try?
